deal/introspection/_extractor.py

```diff
@@ -1,6 +1,8 @@
 from __future__ import annotations
 
-from typing import Callable, Iterator, TypeVar
+from typing import Callable, Iterator, Protocol, TypeVar
+
+from typing_extensions import TypeGuard
```

Comment:
with `__future__` import, `typing_extensions` imports can be placed behind `if TYPE_CHECKING`, making the dependency optional.